tensorflow/lite/micro/tools/make/targets/cortex_m_corstone_300_makefile.inc

```diff
@@ -0,0 +1,180 @@
+endif
+
+ETHOS_U_CORE_PLATFORM := ${PWD}/$(MAKEFILE_DIR)/downloads/ethos_u_core_platform/targets/corstone-300
+DOWNLOAD_RESULT := $(shell $(MAKEFILE_DIR)/ethos_u_core_platform_download.sh ${MAKEFILE_DIR}/downloads)
+ifneq ($(DOWNLOAD_RESULT), SUCCESS)
+  $(error Something went wrong with the Ethos-U Core Platform software download: $(DOWNLOAD_RESULT))
+endif
+
+# This target has dependencies to CMSIS-Device so just in case running without OPTIMIZED_KERNEL_DIR=cmsis_nn.
+CMSIS_DEFAULT_DOWNLOAD_PATH := $(MAKEFILE_DIR)/downloads/cmsis
+CMSIS_PATH := $(CMSIS_DEFAULT_DOWNLOAD_PATH)
+ifeq ($(CMSIS_PATH), $(CMSIS_DEFAULT_DOWNLOAD_PATH))
+  DOWNLOAD_RESULT := $(shell $(MAKEFILE_DIR)/ext_libs/cmsis_download.sh ${MAKEFILE_DIR}/downloads)
+  ifneq ($(DOWNLOAD_RESULT), SUCCESS)
+    $(error Something went wrong with the CMSIS download: $(DOWNLOAD_RESULT))
+  endif
+endif
+
+FLOAT := soft
+GCC_TARGET_ARCH := $(TARGET_ARCH)
+
+ifeq ($(TARGET_ARCH), cortex-m0)
+  CORE=M0
+
+else ifeq ($(TARGET_ARCH), cortex-m3)
+  CORE=M3
+
+else ifeq ($(TARGET_ARCH), cortex-m33)
+  CORE=M33
+  FLOAT=hard
+  CMSIS_ARM_FEATURES := _DSP_DP
+
+else ifeq ($(TARGET_ARCH), cortex-m33+nodsp)
+  CORE=M33
+
+else ifeq ($(TARGET_ARCH), cortex-m4)
+  CORE=M4
+  GCC_TARGET_ARCH := cortex-m4+nofp
+
+else ifeq ($(TARGET_ARCH), cortex-m4+fp)
+  CORE=M4
+  FLOAT=hard
+  GCC_TARGET_ARCH := cortex-m4
+  CMSIS_ARM_FEATURES := _FP
+
+else ifeq ($(TARGET_ARCH), cortex-m55)
+  CORE=M55
+  FLOAT=hard
+
+else ifeq ($(TARGET_ARCH), cortex-m55+nodsp+nofp)
+  CORE=M55
+
+else ifeq ($(TARGET_ARCH), cortex-m55+nofp)
+  CORE=M55
+
+else ifeq ($(TARGET_ARCH), cortex-m7)
+  CORE=M7
+  GCC_TARGET_ARCH := cortex-m7+nofp
+
+else ifeq ($(TARGET_ARCH), cortex-m7+fp)
+  CORE=M7
+  FLOAT=hard
+  GCC_TARGET_ARCH := cortex-m7
+  CMSIS_ARM_FEATURES := _DP
+
+else
+  $(error "TARGET_ARCH=$(TARGET_ARCH) is not supported")
+endif
+
+ifneq ($(filter cortex-m55%,$(TARGET_ARCH)),)
+  # soft-abi=soft disables MVE - use softfp instead for M55.
+  ifeq ($(FLOAT),soft)
+    FLOAT=softfp
+  endif
+endif
+
+ifeq ($(TOOLCHAIN), gcc)
+  export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+  DOWNLOAD_RESULT := $(shell $(MAKEFILE_DIR)/arm_gcc_download.sh ${MAKEFILE_DIR}/downloads)
+  ifneq ($(DOWNLOAD_RESULT), SUCCESS)
+    $(error Something went wrong with the GCC download: $(DOWNLOAD_RESULT))
+  endif
+  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
+
+  FLAGS_GCC = -mcpu=$(GCC_TARGET_ARCH) -mfpu=auto
+  CXXFLAGS += $(FLAGS_GCC)
+  CCFLAGS += $(FLAGS_GCC)
+
+  # Run C preprocessor on linker file to get rid of ifdefs.
+  ifeq (,$(wildcard $(ETHOS_U_CORE_PLATFORM)/platform_parsed.ld))
+    RESULT := $(shell arm-none-eabi-gcc -E -x c -P -o $(ETHOS_U_CORE_PLATFORM)/platform_parsed.ld \
+      $(ETHOS_U_CORE_PLATFORM)/platform.ld)
+    ifneq ($(RESULT),)
+      $(error Something went wrong when parsing linker file: $(RESULT))
+    endif
+  endif
```

Comment:
can this be done as part of the shell script that downloads ethou_core or cornstone instead of the makefile?

We can have the shell script call the arm_gcc download script too, if that helps.

Reply:
Yes will do